ga4gh/backend.py

```diff
@@ -496,15 +496,10 @@ def variantAnnotationSetsGenerator(self, request):
         """
         compoundId = datamodel.VariantSetCompoundId.parse(request.variantSetId)
         dataset = self.getDataRepository().getDataset(compoundId.datasetId)
-        results = []
-        for annset in dataset.getVariantAnnotationSets():
-            try:
-                variantSetId = request.variantSetId
-            except ValueError:
-                variantSetId = ""
-            if str(annset._variantSetId) == str(variantSetId):
-                results.append(annset)
-        return self._objectListGenerator(request, results)
+        variantSet = dataset.getVariantSet(request.variantSetId)
```

Comment:
The schema says nothing on the topic, since there is no data set associated with  the variant annotation set object. So, as far as I'm concerned, this is the only correct interpretation of the schemas. 

I agree that we should flatten the object hierarchy, and translate GET requests directly into SQL requests. This will be done once we've changed the ID system (probably post protobuf, to try and minimise disruption).

If the current code works and doesn't break anything, please move discussion to an issue. We want to include these changes ASAP.